helix-term/src/commands/typed.rs

```diff
@@ -720,8 +735,19 @@ pub fn write_all_impl(
         cx.editor.save::<PathBuf>(id, None, force)?;
     }
 
-    if !errors.is_empty() && !force {
-        bail!("{:?}", errors);
+    if force {
+        return Ok(());
+    }
+
+    let mut error_msg = errors.join(" ");
+    if !readonly_errors.is_empty() {
+        // statusline text will have the form:
+        // ["'readonly' option is set for /home/user/readonly.txt", ...] (add ! to overwrite)
+        error_msg = format!("{:?} (add ! to overwrite)", readonly_errors);
+    }
+
+    if !error_msg.is_empty() && !force {
```

Comment:
```suggestion
    if !error_msg.is_empty() {
```
not needed herer since you bail early above